theano/gpuarray/tests/test_dnn.py

```diff
@@ -2131,3 +2131,64 @@ def funcs(out, params):
                                            (i + 1) * len(cudnn_grads_layer)]
         for j, g in enumerate(cudnn_grads_layer):
             utt.assert_allclose(ref_grads_layer[j], g)
+
+
+def test_dnn_spatialtf_grid_generator():
+    if not dnn.dnn_available(test_ctx_name):
+        raise SkipTest(dnn.dnn_available.msg)
+    utt.seed_rng()
+
+    # shape: (num_images, channels, height, width), equivalent to NCHW
+    grid_dims = (10, 3, 64, 128)
+
+    identity = [[1, 0, 0],
+                [0, 1, 0]]
+
+    theta = np.asarray(grid_dims[0] * [identity], dtype=theano.config.floatX)
+    theta_gpu = gpuarray_shared_constructor(theta)
+
+    from scipy import misc
+    f = misc.face().astype(np.float32)
+    f = np.transpose(f, axes=(2, 0, 1))
+    # Scale input from [0, 255] to [0, 2]
+    sc = 1. / 128.
+    f *= sc
+    # Re-scale input from [0, 2] to [-1, 1] (normalized)
+    f -= 1
+
+    # Create array of images
+    img = np.asarray(grid_dims[0] * [f], dtype=theano.config.floatX)
+    # Create GPU variable for the images
+    img_gpu = gpuarray_shared_constructor(img)
+
+    spatialtf = dnn.dnn_spatialtf(img_gpu, theta_gpu, grid_dims)
+
+    spatialtf_fn = theano.function([], [spatialtf], mode=mode_with_gpu)
+
+    result, = spatialtf_fn()
+
+    img_out = np.asarray(result, dtype=np.float32)
+
+    print(img_out.shape)
+
+    for i in range(len(img_out)):
+        # Re-scale output to range [0, 2]
+        img_out[i] += 1
+        # Re-scale output to range [0, 255]
+        img_out[i] *= 128
+    img_out = img_out.astype(dtype=np.uint8)
+    img_out = np.transpose(img_out, axes=(0, 2, 3, 1))
+
+    for i in range(len(img_out)):
+        print("[sampled image #{0}]".format(i))
+        print("Min/Max: {0}/{1}".format(img_out[i].min(), img_out[i].max()))
+        print(img_out[i])
+
+    import matplotlib.pyplot as plt
+    for img_idx in range(len(img_out)):
+        plt.imshow(img_out[img_idx])
+        plt.show()
+
+    topo = spatialtf_fn.maker.fgraph.toposort()
+    assert len([n for n in topo if isinstance(n.op, dnn.GpuDnnGridGenerator)]) == 1
+    assert len([n for n in topo if isinstance(n.op, dnn.GpuDnnGridSampler)]) == 1
```

Comment:
NB: About that, let's end with CTC pull request and this PR for GPU Spatial Transformation.

After that, we will discuss if your last part of GSoC will be to implement either CPU version for spatial transformation, or something else.